File: connect_db.py
import psycopg2
import json
"""
Allows us to query specific commands
"""
def close(connection):
    #Push the data onto the database
    connection.commit()
                
    print("Data now deleted, close if done!")
    # Close the database
    connection.close()
def get_table():
    with open("settings/dblogin.json", "r") as loop:
                        data = json.load(loop)
                        
    return data["Login"]["Table"]

    
def connect():
    with open("settings/dblogin.json", "r") as loop:
                        data = json.load(loop)
    return psycopg2.connect(user = data["Login"]["Username"],
                                        password = data["Login"]["Password"],
                                        host = data["Login"]["Host"],
                                        port = data["Login"]["Port"],
                                        database = data["Login"]["Database"])    

        

File: backend/v2/src/services/connectDB.go
package services

import (
	"context"
	"encoding/json"
	"fmt"
	"io/ioutil"
	"os"

	"github.com/go-pg/pg/v10"
)

type DB struct {
	Login Login `json:"Login"`
}

type Login struct {
	Username string `json:"Username"`
	Password string `json:"Password"`
	Database string `json:"Database"`
	Host     string `json:"Host"`
	Table    string `json:"Table"`
	Port     string `json:"Port"`
}

func ParseJson() DB {
	info, err := os.Open("settings/dblogin.json")
	if err != nil {
		fmt.Println(err)
	}
	byteValue, _ := ioutil.ReadAll(info)
	var loginJson DB

	jsonErr := json.Unmarshal(byteValue, &loginJson)
	if jsonErr != err {
		fmt.Println(jsonErr)
	}
	return loginJson
}

func GetTable() string {
	json := ParseJson()
	fmt.Println(json.Login.Table)
	return json.Login.Table
}
func CloseDB() {
	db := ConnectToDB()
	defer db.Close()
}

func ConnectToDB() *pg.DB {
	json := ParseJson()
	db := pg.Connect(&pg.Options{
		User:     json.Login.Username,
		Password: json.Login.Password,
		Database: json.Login.Database,
		Addr:     json.Login.Port,
	})
	fmt.Println(db)
	ctx := context.Background()
	if err := db.Ping(ctx); err != nil {
		fmt.Println("Could not connect!")
	}

	return db
}
